src/Core/Service/ServiceRegistry/Client.php

```diff
@@ -42,6 +45,23 @@ public function get(string $name): ServiceEntry
         throw ServiceException::notFound('name', $name);
     }
 
+    /**
+     * @return \Generator<ChunkInterface>
+     */
+    public function fetchServiceZipVersion(string $zipUrl): \Generator
```

Comment:
`fetchServiceZip` version doesn't make much sense here imo. It sounds like fetch just the version info, but we're actually downloading binary zip.